modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistry.java

```diff
@@ -199,6 +203,24 @@ public Object lookupObject(String key)
         }
     }
 
+    private boolean shouldApplyPrototypeLifecycle(String key, Object object)
+    {
+        if (applicationContext.isSingleton(key))
+        {
+            return false;
+        }
+
+        if (object instanceof MessageProcessor)
```

Comment:
Why these types and not others?  At minimum this needs explaining in a comment, but isn't the a bigger issue here?